src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacGraphSketcherValidator.java

```diff
@@ -1,48 +1,161 @@
 import uk.ac.cam.cl.dtg.segue.dos.content.Choice;
 import uk.ac.cam.cl.dtg.segue.dos.content.Content;
 import uk.ac.cam.cl.dtg.segue.dos.content.GraphChoice;
 import uk.ac.cam.cl.dtg.segue.dos.content.Question;
+import uk.ac.cam.cl.dtg.segue.quiz.ISpecifier;
 import uk.ac.cam.cl.dtg.segue.quiz.IValidator;
 import uk.ac.cam.cl.dtg.segue.quiz.ValidatorUnavailableException;
 
+import javax.xml.ws.Response;
 import java.io.IOException;
-import java.io.StringWriter;
-import java.util.*;
+import java.util.Date;
+import java.util.List;
 
 /**
  * Validator that only provides functionality to validate graph questions.
  *
  * Created by hhrl2 on 01/08/2016.
  */
-public class IsaacGraphSketcherValidator implements IValidator {
+public class IsaacGraphSketcherValidator implements IValidator, ISpecifier {
 
     /**
      * Private logger for printing error messages on console.
      */
     private static final Logger log = LoggerFactory.getLogger(IsaacGraphSketcherValidator.class);
 
+    private static final AnswerToInput answerToInput = new AnswerToInput();
+    private static final Features features = new Features();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
     @Override
     public final QuestionValidationResponse validateQuestionResponse(final Question question, final Choice answer) {
-        return new QuestionValidationResponse(question.getId(), answer, false,
-                new Content(answer.getValue() + " (" + Integer.toString(answer.getValue().length()) + ")"), new Date());
+        Validate.notNull(question);
+        Validate.notNull(answer);
+
+        if (!(question instanceof IsaacGraphSketcherQuestion)) {
+            throw new IllegalArgumentException(String.format(
+                "This validator only works with Isaac Graph Sketcher Questions... (%s is not string match)",
+                question.getId()));
+        }
+
+        if (!(answer instanceof GraphChoice)) {
+            throw new IllegalArgumentException(String.format(
+                "Expected GraphChoice for IsaacGraphSketcherQuestion: %s. Received (%s) ", question.getId(),
+                answer.getClass()));
+        }
+
+        IsaacGraphSketcherQuestion graphSketcherQuestion = (IsaacGraphSketcherQuestion) question;
+
+        // These variables store the important features of the response we'll send.
+        Content feedback = null;                        // The feedback we send the user
+        boolean responseCorrect = false;                // Whether we're right or wrong
+
+        if (null == graphSketcherQuestion.getChoices() || graphSketcherQuestion.getChoices().isEmpty()) {
+            log.error("Question does not have any answers. " + question.getId() + " src: "
+                + question.getCanonicalSourceFile());
+
+            feedback = new Content("This question does not have any correct answers");
+        }
+
+        // STEP 1: Did they provide an answer at all?
+
+        GraphAnswer graphAnswer = null;
```

Comment:
I'm just wondering if there is a nicer way of encapsulating the GraphAnswer library do as an isaac Choice / Answer class so that this mapping can be done as the object comes into RESTEasy. 

This is fine for now though; it is just a thought.

Reply:
Possibly. I just took what was already here and we were keen the library could develop separately so that's why the structures are split across the two places a bit.